src/components/Tabs/tests/Tabs.test.tsx

```diff
@@ -193,4 +206,245 @@ describe('<Tabs />', () => {
+      const disclosureWidth = 50;
+      const tabWidths = [82, 160, 150, 100, 80, 120];
+      const containerWidth = 300;
+      const actualIndices = getVisibleAndHiddenTabIndices(
+        mockTabs,
+        selected,
+        disclosureWidth,
+        tabWidths,
+        containerWidth,
+      );
+      const expectedIndices = {visibleTabs: [0, 1], hiddenTabs: [2, 3, 4, 5]};
+      expect(actualIndices).toEqual(expectedIndices);
+    });
+  });
+
+  describe('<Popover />', () => {
+    it('renders a Popover with preferredPosition below when there are hiddenTabs', () => {
+      const mockTabs = [
+        {content: 'Tab 1', id: 'tab-1'},
+        {content: 'Tab 2', id: 'tab-2'},
+        {content: 'Tab 3', id: 'tab-3'},
+        {content: 'Tab 4', id: 'tab-4'},
+        {content: 'Tab 5', id: 'tab-5'},
+        {content: 'Tab 6', id: 'tab-6'},
+      ];
+
+      const tabs = mountWithAppProvider(
+        <Tabs selected={0} tabs={mockTabs} onSelect={noop} />,
+      );
+      tabs.setState({
+        tabWidths: [82, 160, 150, 100, 80, 120],
+        containerWidth: 300,
+      });
+
+      const popover = tabs.find(Popover);
+      expect(popover).toHaveLength(1);
+      expect(popover.prop('preferredPosition')).toBe('below');
+    });
+
+    it('renders a Popover with a button as the activator when there are hiddenTabs', () => {
+      const mockTabs = [
+        {content: 'Tab 1', id: 'tab-1'},
+        {content: 'Tab 2', id: 'tab-2'},
+        {content: 'Tab 3', id: 'tab-3'},
+        {content: 'Tab 4', id: 'tab-4'},
+        {content: 'Tab 5', id: 'tab-5'},
+        {content: 'Tab 6', id: 'tab-6'},
+      ];
+
+      const tabs = mountWithAppProvider(
+        <Tabs selected={0} tabs={mockTabs} onSelect={noop} />,
+      );
+      tabs.setState({
+        tabWidths: [82, 160, 150, 100, 80, 120],
+        containerWidth: 300,
+      });
+
+      const popover = tabs.find(Popover);
+      expect(popover.prop('activator').type).toBe('button');
+    });
+
+    describe('ArrowRight', () => {
+      xit('shifts focus to the last tab when pressing ArrowRight', () => {
+        const mockTabs = [
+          {content: 'Tab 1', id: 'tab-1'},
+          {content: 'Tab 2', id: 'tab-2'},
+          {content: 'Tab 3', id: 'tab-3'},
+          {content: 'Tab 4', id: 'tab-4'},
+          {content: 'Tab 5', id: 'tab-5'},
+          {content: 'Tab 6', id: 'tab-6'},
+        ];
+
+        const tabs = mountWithAppProvider(
+          <Tabs selected={0} tabs={mockTabs} onSelect={noop} />,
+        );
+        tabs.setState({
+          tabWidths: [82, 160, 150, 100, 80, 120],
+          containerWidth: 300,
+        });
+
+        const popoverContents = getPopoverContents(tabs);
+        trigger(popoverContents.find(List), 'onKeyPress', {
+          key: 'ArrowRight',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+    });
+  });
 });
+
+function getPopoverContents(tabs: ReactWrapper) {
```

Comment:
I know we do this in components but in tests for readability I would define this just below your `mockProps` definition.